src/target/riscv/riscv-013.c

```diff
@@ -3720,15 +3750,19 @@ read_memory_abstract(struct target *target, const riscv_mem_access_args_t args)
 	assert(riscv_mem_access_is_read(args));
 
 	RISCV013_INFO(info);
-	bool use_aampostincrement = info->has_aampostincrement != YNM_NO;
-
 	memset(args.read_buffer, 0, args.count * args.size);
 
 	/* Convert the size (bytes) to width (bits) */
 	unsigned int width = args.size << 3;
 
 	/* Create the command (physical address, postincrement, read) */
-	uint32_t command = access_memory_command(target, false, width, use_aampostincrement, false);
+	uint32_t command = access_memory_command(target, false, width,
+			/* postincrement = */ true, /* is_write= */ false);
+	bool use_aampostincrement =
+			!ac_cache_contains(&info->ac_not_supported_cache, command);
+	if (!use_aampostincrement)
+		command = access_memory_command(target, false, width,
+				/* postincrement = */ false, /* is_write= */ false);
```

Comment:
You can simplify this part of the code:

```suggestion
	bool use_aampostincrement =
			!ac_cache_contains(&info->ac_not_supported_cache, command);
	
	uint32_t command = access_memory_command(target, false, width,
			use_aampostincrement, /* is_write= */ false);	
```

Reply:
Addressed